### PR TITLE
Introduce final_width&height for pixelpipes

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Currently there is no way to find out the final dimensions of a pixelpipe while processing it. This introduces final_width& height in the pipe struct taht can be read by modules while processing helping to possibly find optimal processing code.

While being here also fixed static functions styles.

### CODE REVIEW

File: src/develop/pixelpipe_hb.c [hunk 1]
```diff
@@ -2243,6 +2245,8 @@ int dt_dev_pixelpipe_process(dt_dev_pixelpipe_t *pipe, dt_develop_t *dev, int x,
   if(pipe->devid >= 0) dt_opencl_events_reset(pipe->devid);
 
   dt_iop_roi_t roi = (dt_iop_roi_t){ x, y, width, height, scale };
+  pipe->final_width = width;
+  pipe->final_height = height;
```
Comment (TurboGit) [inline src/develop/pixelpipe_hb.c:2249]: A bit later in this routine we have:

```
  // terminate
  dt_pthread_mutex_lock(&pipe->backbuf_mutex);
  pipe->backbuf_hash = dt_dev_pixelpipe_cache_hash(pipe->image.id, &roi, pipe, 0);
  pipe->backbuf = buf;
  pipe->backbuf_width = width;
  pipe->backbuf_height = height;
```

So to me backbuf_width/height is final_width/height, no?

Comment (jenshannoschwalm) [reply src/develop/pixelpipe_hb.c:2250]: Unfortunately not, further down while processing the pipeline this holds dimensions of a valid display buffer, this is zero while exporting to file/thumbnail.

just pushed commit displays data via -d roi while processing the module...

**Comment by TurboGit** [comment]
This is breaking a CI:

```
In file included from D:/a/darktable/darktable/src/src/develop/pixelpipe.c:22:
D:/a/darktable/darktable/src/src/develop/pixelpipe_hb.c:67:13: error: conflicting types for '_get_output_format'; have 'void(dt_iop_module_t *, dt_dev_pixelpipe_t *, dt_dev_pixelpipe_iop_t *, dt_develop_t *, dt_iop_buffer_dsc_t *)'
   67 | static void _get_output_format(dt_iop_module_t *module, dt_dev_pixelpipe_t *pipe, dt_dev_pixelpipe_iop_t *piece,
      |             ^~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.

[7/444] D:\a\_temp\msys64\ucrt64\bin\gcc.exe -DGDK_DISABLE_DEPRECATED -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_22 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_MIN_REQUIRED 
```

**Comment by jenshannoschwalm** [comment]
Two adventures here. Leading underscore in static function names and git commit squashing :-)

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**